core/src/avm2/globals/flash/filters/ColorMatrixFilter.as

```diff
@@ -0,0 +1,12 @@
+﻿package flash.filters {
+	public final class ColorMatrixFilter extends BitmapFilter {
+		public var matrix: Array;
+
+		public function ColorMatrixFilter(matrix: Array = null) {
+			this.matrix = matrix;
+		}
+		override public function clone(): BitmapFilter {
+			return new ColorMatrixFilter(this.matrix.concat());
```

Comment:
Why `.concat()`?

Reply:
Calling `concat()` makes a shallow copy (which is fine here, since `this.matrix` should contain numbers). Otherwise, we'd end up with two `ColorMatrixFilters` with the same underlying array.

Reply:
This is still not 100% accurate, since `.matrix` should be an array of 20 items for 4 x 5 color transform matrix (See [documentation](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/filters/ColorMatrixFilter.html#matrix)).

Consider the following AS3 code:
```as
import flash.filters.*;
var a = [1, 2, 3];
var f = new ColorMatrixFilter(a);
trace(f.matrix); // traces "1,2,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0" - an array of 20 items
```

Also, note that arrays containing more than 20 items are clamped:
```as
import flash.filters.*;
var a = [1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,3];
var f = new ColorMatrixFilter(a);
trace(f.matrix); // traces "1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2" - an array of 20 items
```

And that's not all - The items should be coerced to `f64` upon a `ColorMatrixFilter` creation:
```as
import flash.filters.*;
var a = ["1.5"];
var f = new ColorMatrixFilter(a);
trace(typeof f.matrix[0]); // traces "number" (which is 1.5)
```

In short, it's very likely that Flash stored the matrix data in `[f64; 20]` (or even better, `[[f64; 5]; 4]`), so I highly recommend doing so in Ruffle too, once we have native bindings (#7244).